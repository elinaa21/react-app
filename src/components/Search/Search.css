.search {
    position: relative;
    margin-top: 10%;
    margin-bottom: 10%;
    height: 5%;
    text-align: center;
    display: flex;
    justify-content: center;
}

.search__input {
    font-family: 'Gill Sans', sans-serif;
    width: 70%;
    outline: none;
    border-radius: 5px 0 0 5px;
    background-color: rgba(255, 255, 255, 0.3);
    border: none;
    font-size: 100%;
    color: rgba(255, 255, 255, 0.829);
    padding-left: 8%;
}

.search__input::placeholder {
    color: rgba(255, 255, 255, 0.829);
}

.search__img {
    width: 12%;
    border-radius: 0 5px 5px 0;
    filter: invert(1);
    background-color: rgba(0, 0, 0, 0.3);
    background-image: url('../../icons/search-icon.png');
    background-position: center;
    background-repeat: no-repeat;
    background-size: 60%;
}