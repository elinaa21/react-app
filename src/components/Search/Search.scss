.search {
    position: fixed;
    margin-top: 2%;
    height: 5%;
    text-align: center;
    display: flex;
    justify-content: center;
    width: 20%;

    &__input {
        font-family: 'Gill Sans', sans-serif;
        width: 70%;
        outline: none;
        border-radius: 5px 0 0 5px;
        background-color: rgba(255, 255, 255, 0.3);
        border: none;
        font-size: 100%;
        color: rgba(255, 255, 255, 0.829);
        padding-left: 8%;

        &::placeholder {
            color: rgba(255, 255, 255, 0.829);
        }
    }

    &__img {
        width: 12%;
        border-radius: 0 5px 5px 0;
        filter: invert(1);
        background: {
            color: rgba(0, 0, 0, 0.3);
            image: url('../../icons/search-icon.png');
            position: center;
            repeat: no-repeat;
            size: 60%;
        }
    }
}