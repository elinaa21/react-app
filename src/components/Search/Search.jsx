import React from 'react';
import './Search.css';


const Search = () => (
    <div className='search'>
        <input type='text' className='search__input' placeholder='поиск' />
        <div className='search__img' />
    </div>
);

export default Search;
