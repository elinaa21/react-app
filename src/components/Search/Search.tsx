import React from 'react';

import { cn } from '../../modules/cn';

import './Search.scss';


const Search: React.FC = () => (
    <div className = {cn('search')}>
        <input type='text' className = {cn('search', 'input')} placeholder='search' />
        <div className = {cn('search', 'img')} />
    </div>
);

export default Search;
