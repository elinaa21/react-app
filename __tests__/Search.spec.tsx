import * as React from 'react';
import * as renderer from 'react-test-renderer';
import Search from '../src/components/Search/Search';

it('SubjectToBeTested renders correctly', () => {
  const tree = renderer
    .create(<Search />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
