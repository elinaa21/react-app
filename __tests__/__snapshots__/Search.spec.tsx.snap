// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SubjectToBeTested renders correctly 1`] = `
<div
  className="search"
>
  <input
    className="search__input"
    placeholder="search"
    type="text"
  />
  <div
    className="search__img"
  />
</div>
`;
